File: patterns-structures/composite/code-java/type_fichier/FichierTxt.java
package type_fichier;

import fichier.Fichier;

public class FichierTxt extends Fichier {

    private String type;

    public FichierTxt(String nom) {
        super(nom);

        this.type = "txt";
    }

    @Override
    public String getType(){
        return type;
    }

    @Override
    public void decrire() { 
        System.out.println("Nom: " + super.getNom() + "\nType: " + type);
    }
}

File: 2-pattrons-de-structures/composite/code-typescript/type_fichier/fichierTxt.ts
import { Fichier } from "../fichier/fichier.js";

export class FichierTxt extends Fichier {
  constructor(protected nom: string, private type: string = "txt") {
    super(nom);
  }

  getType(): string {
    return this.type;
  }

  decrire(): void {
    console.log(`Nom: ${this.getNom()}, Type: ${this.getType()}`);
  }
}
